warehouse/transformer/events.go

```diff
@@ -164,42 +164,31 @@ func (t *Transformer) trackEventsResponse(tec *transformEventContext, transforme
 }
 
 func (t *Transformer) extractEvents(tec *transformEventContext) ([]map[string]any, error) {
-	data := make(map[string]any)
-	metadata := make(map[string]string)
-
-	if err := setDataAndMetadataFromInput(tec, t.eventContext(tec), data, metadata, &prefixInfo{
-		completePrefix: "extract_context_",
-		completeLevel:  2,
-		prefix:         "context_",
-	}); err != nil {
-		return nil, fmt.Errorf("extract: setting data and column types from input: %w", err)
-	}
-	if err := setDataAndMetadataFromInput(tec, tec.event.Message["properties"], data, metadata, &prefixInfo{
-		completePrefix: "extract_properties_",
-		completeLevel:  2,
-	}); err != nil {
-		return nil, fmt.Errorf("extract: setting data and column types from input: %w", err)
+	commonData, commonMetadata, err := t.extractCommonProps(tec)
+	if err != nil {
+		return nil, fmt.Errorf("identifies: common properties: %w", err)
 	}
 
 	eventColName, err := safeColumnNameCached(tec, "event")
 	if err != nil {
 		return nil, fmt.Errorf("extract: safe column name: %w", err)
 	}
 
-	eventName, _ := tec.event.Message[eventColName].(string)
-	transformedEventName := transformTableNameCached(tec, eventName)
-	if utils.IsBlank(transformedEventName) {
-		return nil, response.ErrExtractEventNameEmpty
-	}
-
-	data[eventColName] = transformedEventName
-	metadata[eventColName] = model.StringDataType
+	extractData := make(map[string]any)
+	extractMetadata := make(map[string]string)
 
-	if err = setDataAndMetadataFromRules(tec, data, metadata, rules.ExtractRules); err != nil {
-		return nil, fmt.Errorf("extract: setting data and column types from rules: %w", err)
+	if err := setDataAndMetadataFromInput(tec, tec.event.Message["properties"], extractData, extractMetadata, &prefixInfo{
+		completePrefix: "extract_properties_",
+		completeLevel:  2,
+	}); err != nil {
+		return nil, fmt.Errorf("extract: setting data and column types from input: %w", err)
 	}
 
-	columnName := transformColumnNameCached(tec, transformedEventName)
+	data := lo.Assign(extractData, commonData)
+	metadata := lo.Assign(extractMetadata, commonMetadata)
```

Comment:
Why this assignment?

Reply:
We need to club both the commonData and extractData to get the final data based on priority for commonData. That's why it was kept at the end.